bot/cogs/watchchannels/watchchannel.py

```diff
@@ -0,0 +1,359 @@
+        return True
+
+    async def fetch_user_cache(self) -> bool:
+        """
+        Fetches watched users from the API and updates the watched user cache accordingly.
+
+        This function returns `True` if the update succeeded.
+        """
+        try:
+            data = await self.bot.api_client.get(
+                self.api_endpoint,
+                params=self.api_default_params
+            )
+        except aiohttp.ClientResponseError as e:
+            self.log.exception(
+                f"Failed to fetch {self.__class__.__name__} watched users from API",
+                exc_info=e
+            )
+            return False
+
+        self.watched_users = defaultdict(dict)
+
+        for entry in data:
+            user_id = entry.pop('user')
+            self.watched_users[user_id] = entry
+
+        return True
+
+    async def on_message(self, msg: Message) -> None:
+        """Queues up messages sent by watched users."""
+        if msg.author.id in self.watched_users:
+            if not self.consuming_messages:
+                self._consume_task = self.bot.loop.create_task(self.consume_messages())
+
+            self.log.trace(f"Received message: {msg.content} ({len(msg.attachments)} attachments)")
+            self.message_queue[msg.author.id][msg.channel.id].append(msg)
+
+    async def consume_messages(self, delay_consumption: bool = True) -> None:
+        """Consumes the message queues to log watched users' messages."""
+        if delay_consumption:
+            self.log.trace(f"Sleeping {BigBrotherConfig.log_delay} seconds before consuming message queue")
+            await asyncio.sleep(BigBrotherConfig.log_delay)
+
+        self.log.trace(f"{self.__class__.__name__} started consuming the message queue")
+
+        # Prevent losing a partly processed consumption queue after Task failure
+        if not self.consumption_queue:
+            self.consumption_queue = self.message_queue.copy()
+            self.message_queue.clear()
+
+        for user_channel_queues in self.consumption_queue.values():
+            for channel_queue in user_channel_queues.values():
+                while channel_queue:
+                    msg = channel_queue.popleft()
+
+                    self.log.trace(f"Consuming message {msg.id} ({len(msg.attachments)} attachments)")
+                    await self.relay_message(msg)
+
+        self.consumption_queue.clear()
+
+        if self.message_queue:
+            self.log.trace("Channel queue not empty: Continuing consuming queues")
+            self._consume_task = self.bot.loop.create_task(
+                self.consume_messages(delay_consumption=False)
+            )
+        else:
+            self.log.trace("Done consuming messages.")
+
+    async def webhook_send(
+        self, content: Optional[str] = None, username: Optional[str] = None,
+        avatar_url: Optional[str] = None, embed: Optional[Embed] = None,
```

Comment:
I really don't want to hold up the PR with style shit but I think our preferred style is to put each parameter on a separate line when the signature is too long. If no consensus we'll just take care of this later.

Reply:
That's an interesting point; I'm not sure what the current consensus is. Both are styles, let's call them parameter Tetris and one-per-line style, are prevalent in our bot code base. 

Here are some examples of the Tetris style:

https://github.com/python-discord/bot/blob/django/bot/pagination.py#L96

https://github.com/python-discord/bot/blob/django/bot/pagination.py#L317

Here, the one-per-line style (more prevalent in django-branch):

https://github.com/python-discord/bot/blob/5e16f4a52d59c73a04323e070e7b4a320e8c1e49/bot/utils/messages.py#L15

https://github.com/python-discord/bot/blob/5e16f4a52d59c73a04323e070e7b4a320e8c1e49/bot/rules/attachments.py#L8

I'm leaning towards the one-per-line style, so I'm going to change it here (since both seem acceptable). There was debate surrounding black recently in core-dev, though, that touched on similar issues (tetris vs one-per-line) that had people arguing for both sides.

Reply:
I've changed this to the a `one-per-line`-style.